src/components/Clock.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import './Clock.scss';
+
+type Props = {
+  name: string | null;
+};
+
+type State = {
+  date: Date;
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
```

Comment:
```suggestion
  state: State = {
```

Reply:
do we have to that even though we pass State into the component as generic? i tried modifying the state object, but it threw an error that the state was not of the same shape as the passed State. so i just assumed that state: State is somehow done by default

Reply:
It's not strictly necessary. But better to annotate it.
You can read something about it here https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/class_components/

Reply:
gotcha. thank you!